### PR TITLE
Refactor thruster control

### PR DESCRIPTION
**Pull request by patapufate**
This Patch:

- Remove unnecessary boilerplate in CMakeList.txt and Package.xml.
- Remove commented/obsolete code
- Add mantainer and license.
- Add Roslint and fix linter errors.
- Change folder structure for header files. Now implemented as "include/<package_name>/header_file.h".

It needs thorough testing, as I don't have the hardware at hand.

### CODE REVIEW

File: thruster_control/include/thruster_control/CO_VehicleSBC.h [hunk 1]
```diff
@@ -47,25 +47,25 @@ const indextable * CO_VehicleSBC_scanIndexOD (CO_Data *d, UNS16 wIndex, UNS32 *
...
+extern UNS8 BootloaderMajorVersion;     /* Mapped at index 0x2024, subindex 0x00*/
+extern UNS8 BootloaderCommonVersion;    /* Mapped at index 0x2025, subindex 0x00*/
+extern UNS8 BootloaderMinorVersion;     /* Mapped at index 0x2026, subindex 0x00*/
+extern UNS8 BootloaderBuildVersion;     /* Mapped at index 0x2027, subindex 0x00*/
+extern UNS8 BITStatus;                  /* Mapped at index 0x2044, subindex 0x00*/
+extern UNS8 platformStateStatus;        /* Mapped at index 0x2051, subindex 0x00*/
+extern UNS16 MotorControllerHeartbeat;  /* Mapped at index 0x2060, subindex 0x00*/
 
-#endif // CO_VEHICLESBC_H
+#endif // THRUSTER_CONTROL_CO_VEHICLESBC_H
```
Comment (hidmic) [inline thruster_control/include/thruster_control/CO_VehicleSBC.h:71]: @patapufate this is a generated file. I'd rather not touch it. It'll make diffs on update larger than necessary.

File: thruster_control/src/thruster_control.cpp [hunk 2]
```diff
@@ -105,65 +107,85 @@ ThrusterControl::ThrusterControl(ros::NodeHandle& nodeHandle)
...
-          },
-          diagnostic_tools::MessageStagnationCheckParams{}.stagnation_diagnostic(
-              {diagnostic_tools::Diagnostic::WARN,
-               health_monitor::ReportFault::THRUSTER_TEMP_STAGNATED})));
+          }
+          , diagnostic_tools::MessageStagnationCheckParams{}
+          .stagnation_diagnostic(
+              {
+                diagnostic_tools::Diagnostic::WARN, ReportFault::THRUSTER_TEMP_STAGNATED
+              }
```
Comment (hidmic) [inline thruster_control/src/thruster_control.cpp:150]: @patapufate nit: perhaps re-arranging this temporaries like we did for the `battery_monitor` is best (for readability).

File: thruster_control/include/thruster_control/CO_VehicleSBC.h (hunk 1 above)
Comment (patapufate) [reply thruster_control/include/thruster_control/CO_VehicleSBC.h:71]: Ok. Fixed

File: thruster_control/src/thruster_control.cpp (hunk 2 above)
Comment (patapufate) [reply thruster_control/src/thruster_control.cpp:150]: OK

File: thruster_control/CMakeLists.txt [hunk 3]
```diff
@@ -193,51 +79,26 @@ target_include_directories(${NODE_APP_NAME} PUBLIC
     pthread
  )
 
+roslint_cpp(
+  src/thruster_control.cpp
+  src/thruster_control_main.cpp
+  include/thruster_control/thruster_control.h
+)
+
+roslint_add_test()
```
Comment (hidmic) [inline thruster_control/CMakeLists.txt:88]: @patapufate same about putting lint tests into the `## Testing ##` block.

File: thruster_control/include/thruster_control/CANIntf.h [hunk 4]
```diff
@@ -93,13 +93,13 @@ class CANIntf
...
-		bool enableCANBusLogging;
+        bool enableCANBusLogging;
 
-		std::vector<std::string> canNodeIdList;
+        std::vector<std::string> canNodeIdList;
 
 };
 
-#endif // _DATA_OBJECT_H
+#endif // THRUSTER_CONTROL_DATA_OBJECT_H
```
Comment (hidmic) [inline thruster_control/include/thruster_control/CANIntf.h:105]: @patapufate did you mean
```suggestion
#endif // THRUSTER_CONTROL_CAN_INTERFACE_H
```
?

File: thruster_control/include/thruster_control/thruster_control.h [hunk 5]
```diff
@@ -34,39 +34,35 @@
...
 #include <diagnostic_updater/diagnostic_updater.h>
-
+#include <health_monitor/ReportFault.h>
 #include <thruster_control/ReportMotorTemperature.h>
 #include <thruster_control/ReportRPM.h>
 #include <thruster_control/SetRPM.h>
-
-#include <health_monitor/ReportFault.h>
-
 #include "CANIntf.h"
```
Comment (hidmic) [inline thruster_control/include/thruster_control/thruster_control.h:55]: @patapufate shouldn't this be 
```suggestion
#include "thurster_control/CANIntf.h"
```
? Also, should we be installing this? I wonder if we shouldn't put these implementation details under `src` (and keep `src` as a private build time include directory).

File: thruster_control/package.xml [hunk 6]
```diff
@@ -4,72 +4,28 @@
...
+  <exec_depend>rospy</exec_depend>
+  <exec_depend>std_msgs</exec_depend>
 
-  <!-- The export tag contains other, unspecified, tags -->
-  <export>
-    <!-- Other tools can request additional information be placed here -->
-
-  </export>
 </package>
+
```
Comment (hidmic) [inline thruster_control/package.xml:31]: @patapufate nit: unnecessary blank line?

File: thruster_control/src/thruster_control.cpp [hunk 7]
```diff
@@ -105,65 +107,85 @@ ThrusterControl::ThrusterControl(ros::NodeHandle& nodeHandle)
...
-                            health_monitor::ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED}
-              .description(
-                  "Temperature above threshold: %f degC > %f degC",
-                  temperature, motorTemperatureThreshold);
+          return Diagnostic
+          {
+            Diagnostic::ERROR, ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED
+          }
+              .description("Temperature above threshold: %f degC > %f degC", temperature,
+                           motorTemperatureThreshold);
```
Comment (hidmic) [inline thruster_control/src/thruster_control.cpp:184]: @patapufate meta nit: I'm starting to think `roslint` was configured for pre-C++11 code. Something like: 
```suggestion
          return Diagnostic(Diagnostic::ERROR, ReportFault::THRUSTER_TEMP_THRESHOLD_REACHED)
              .description("Temperature above threshold: %f degC > %f degC", temperature,
                           motorTemperatureThreshold);
```
would read better and work equally well. Same elsewhere. These "open" uniform initialization statements are awful.

**Review by hidmic — COMMENTED** [review]
Left a few more comments, it's looking better !

File: thruster_control/CMakeLists.txt (hunk 3 above)
Comment (patapufate) [reply thruster_control/CMakeLists.txt:88]: Fixed

File: thruster_control/include/thruster_control/CANIntf.h (hunk 4 above)
Comment (patapufate) [reply thruster_control/include/thruster_control/CANIntf.h:105]: Fixed

File: thruster_control/include/thruster_control/thruster_control.h (hunk 5 above)
Comment (patapufate) [reply thruster_control/include/thruster_control/thruster_control.h:55]: Fixed. Added "Thruster_Control"

File: thruster_control/package.xml (hunk 6 above)
Comment (patapufate) [reply thruster_control/package.xml:31]: Fixed

File: thruster_control/src/thruster_control.cpp (hunk 7 above)
Comment (patapufate) [reply thruster_control/src/thruster_control.cpp:184]: Ok

File: thruster_control/package.xml (hunk 6 above)
Comment (hidmic) [reply thruster_control/package.xml:31]: You sure?

**Review by hidmic — APPROVED** [review]
LGTM

**Pull request merged**